src/App/App.xaml.cs

```diff
@@ -197,6 +203,27 @@ private async Task CheckPasswordlessLoginRequestsAsync()
             }
         }
 
+        private async Task<bool> CheckShouldSwitchActiveUserAsync(PasswordlessRequestNotification notification)
+        {
+            var activeUserId = await _stateService.GetActiveUserIdAsync();
+            var notificationUserEmail = await _stateService.GetEmailAsync(activeUserId);
```

Comment:
This should be after the `if`. You might be doing async work that could not be necessary in the scenario the notification user id is the same as the active one